telegram/ext/filters.py

```diff
@@ -171,6 +173,33 @@ def filter(self, message):
     command = _Command()
     """:obj:`Filter`: Messages starting with ``/``."""
 
+    class regex(BaseFilter):
+        """
+        Filters updates by searching for an occurence of ``pattern`` in the message text.
+        The ``re.search`` function is used to determine whether an update should be filtered.
+        Refer to the documentation of the ``re`` module for more information.
+
+        Note: Does not allow passing groups or a groupdict like the ``RegexHandler`` yet,
+        but this will probably be implemented in a future update, gradually phasing out the
+        RegexHandler (see https://github.com/python-telegram-bot/python-telegram-bot/issues/835).
+
+        Examples:
+            Example ``CommandHandler("start", deep_linked_callback, Filters.regex('parameter'))``
+
+        Args:
+            pattern (:obj:`str` | :obj:`Pattern`): The regex pattern.
+        """
+
+        def __init__(self, pattern):
+            if isinstance(pattern, string_types):
+                pattern = re.compile(pattern)
+            self.pattern = pattern
+            self.name = 'Filters.regex({})'.format(self.pattern)
+
+        def filter(self, message):
+            return bool(re.search(self.pattern, message.text))
```

Comment:
Is there a reason not to do:
 ```python
return bool(self.pattern.search(message.text))
```

Reply:
self.pattern can either be a regex compiled pattern or a string. You're solutions doesn't work for strings.

Reply:
This should be okay